addons/website_slides_survey/models/slide_channel.py

```diff
@@ -2,9 +2,29 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import fields, models
+from odoo.osv import expression
 
 
 class Channel(models.Model):
     _inherit = 'slide.channel'
 
     nbr_certification = fields.Integer("Number of Certifications", compute='_compute_slides_statistics', store=True)
+
+    def _remove_membership(self, partner_ids):
+        """Unlink user_input from the slide.
+
+        Unlinking the user_input from the slide allows to keep track of the current
+        pool of attempts allowed since the user (last) joined the course, as only
+        those will have a slide_id."""
+        removed_channel_partner_domain = []
+        for channel in self:
+            removed_channel_partner_domain = expression.OR([
+                removed_channel_partner_domain,
+                [('partner_id', 'in', partner_ids),
+                 ('channel_id', '=', channel.id)]
+            ])
+        if (removed_channel_partner_domain):
+            slide_partner_ids = self.env['slide.slide.partner'].sudo().search(
```

Comment:
It is short-lived but let it be obvious if someone works on this later.
```suggestion
            slide_partner_ids_sudo = self.env['slide.slide.partner'].sudo().search(
```